src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php

```diff
@@ -68,7 +68,7 @@ public function getTypeExtensions($name)
 
                 // validate the result of getExtendedTypes()/getExtendedType() to ensure it is consistent with the service definition
                 if (!\in_array($name, $extendedTypes, true)) {
-                    throw new InvalidArgumentException(sprintf('The extended type specified for the service "%s" does not match the actual extended type. Expected "%s", given "%s".', $serviceId, $name, implode(', ', $extendedTypes)));
+                    throw new InvalidArgumentException(sprintf('The extended type "%s" specified for the type extension class "%s" does not match any of the actual extended types (["%s"]).', $name, \get_class($extension), implode('", "', $extendedTypes)));
```

Comment:
This error message is a bit confusing to me. Could we use the common Symfony pattern for errors like this? (Maybe it doesn't make sense, I don't know!) E.g.

```
The extended type of class "%s" must be one of ["%s"], "%s" given.
```

Reply:
The suggested wording conveys the assumption that the class returns unexpected data while the extension class currently assumes that the service configuration is incorrect. I don't know what would be better. 🤷‍♂️

Reply:
OK, let's keep then message then. Thanks!